### PR TITLE
Feature/3790/wrong GitHub app language

### PR DESCRIPTION
**Pull request by garyluu**
For #3790

### CODE REVIEW

**Review by denis-yuen — COMMENTED** [review]
Any idea what is going on with the total code coverage failure?

**Comment by garyluu** [comment]
> Any idea what is going on with the total code coverage failure?

I restarted some Travis jobs at some point, I guess Codacy doesn't like it.  I re-pushed, we'll see what happens

**Review by denis-yuen — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Review by wshands — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]

**Review by Ldcabansay — APPROVED** [review]

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903&bid=20290588)

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java [hunk 1]
```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
...
+        } catch (io.dockstore.openapi.client.ApiException e) {
+            Assert.assertEquals("Cannot change descriptor type of a valid workflow", e.getMessage());
+        }
+        PublishRequest publishRequest = SwaggerUtility.createPublishRequest(true);
+        client.publish(workflow.getId(), publishRequest);
+        try {
+            workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.WDL.toString());
+            fail("Should also not be able to change the descriptor type of a workflow that is published");
+        } catch (io.dockstore.openapi.client.ApiException e) {
+            Assert.assertEquals("Cannot change descriptor type of a published workflow", e.getMessage());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java:421]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java [hunk 2]
```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
...
+        Workflow workflow = client.getWorkflowByPath("github.com/" + workflowRepo + "/foobar", "", false);
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version", 1, workflow.getWorkflowVersions().size());
+        assertFalse("Should be invalid (wrong language, bad version)", workflow.getWorkflowVersions().get(0).isValid());
+
+        workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.CWL.toString());
+        io.dockstore.openapi.client.model.Workflow updatedWorkflowAfterModifyingDescriptorType = workflowsApi.getWorkflow(workflow.getId(), "");
+        Assert.assertEquals("The descriptor language should have been changed", io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum.CWL, updatedWorkflowAfterModifyingDescriptorType.getDescriptorType());
+        Assert.assertEquals("The old versions should have been removed", 0, updatedWorkflowAfterModifyingDescriptorType.getWorkflowVersions().size());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java:391]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java [hunk 3]
```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
...
+
+        workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.WDL.toString());
+        updatedWorkflowAfterModifyingDescriptorType = workflowsApi.getWorkflow(workflow.getId(), "");
+        Assert.assertEquals("The descriptor language should have been changed", io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum.WDL, updatedWorkflowAfterModifyingDescriptorType.getDescriptorType());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java:395]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java [hunk 4]
```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
...
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version 0.1", 1, workflow.getWorkflowVersions().size());
+        assertTrue("Should be valid", workflow.getWorkflowVersions().get(0).isValid());
+        try {
+            workflowsApi
+                    .updateDescriptorType(workflow.getId(), DescriptorLanguage.CWL.toString());
+            fail("Should not be able to change the descriptor type of a workflow that has valid versions");
+        } catch (io.dockstore.openapi.client.ApiException e) {
+            Assert.assertEquals("Cannot change descriptor type of a valid workflow", e.getMessage());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java:413]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java [hunk 5]
```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
...
+        // Ensure that new workflow is created and is what is expected
+        Workflow workflow = client.getWorkflowByPath("github.com/" + workflowRepo + "/foobar", "", false);
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version", 1, workflow.getWorkflowVersions().size());
+        assertFalse("Should be invalid (wrong language, bad version)", workflow.getWorkflowVersions().get(0).isValid());
+
+        workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.CWL.toString());
+        io.dockstore.openapi.client.model.Workflow updatedWorkflowAfterModifyingDescriptorType = workflowsApi.getWorkflow(workflow.getId(), "");
+        Assert.assertEquals("The descriptor language should have been changed", io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum.CWL, updatedWorkflowAfterModifyingDescriptorType.getDescriptorType());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java:390]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903&bid=20290588)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/MyWorkflows.java  -95
- dockstore-webservice/src/main/java/io/dockstore/webservice/CustomWebApplicationException.java  -75
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/TRSListener.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/LabelDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/VersionDAO.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TokenDAO.java  -47
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java  -68
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java  -42
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java  -12
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryStarredSerializer.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java  -4
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventSearchType.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java  -62
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java  -58
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/FileFormatDAO.java  -85
- dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java  -96
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java  -92
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dag/ElementsDefinition.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TagDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Validation.java  -59
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java  -51
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/User.java  -54
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PersistenceExceptionMapper.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthorizer.java  -20
- dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java  -80
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java  -35
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Notification.java  -3
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DAGHelper.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/tooltester/ToolTesterLogType.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextflowHandler.java  -88
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitLabSourceCodeRepo.java  -72
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApiService.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Checksum.java  -65
- dockstore-webservice/src/main/java/io/openapi/api/impl/ServiceInfoApiServiceImpl.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/NotificationResource.java  -16
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PublicStateManager.java  -85
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java  -46
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java  -82
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java  -86
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java  -85
- dockstore-webservice/src/main/java/io/dockstore/webservice/CharsetResponseFilter.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Workflow.java  -94
- dockstore-webservice/src/main/java/io/swagger/api/ToolClassesApiV1.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java  -19
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TokenType.java  -93
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TemplateHealthCheck.java  -42
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Label.java  -50
- dockstore-common/src/main/java/io/dockstore/common/NextflowUtilities.java  -73
- dockstore-common/src/main/java/io/dockstore/common/EntryType.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ParsedInformation.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ToolTesterResource.java  -16
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/DescriptorTypeConverter.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Service.java  -14
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dockerhub/Results.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ToolMode.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java  -82
- dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthenticator.java  -61
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsApiServiceImpl.java  -72
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ImageRegistryFactory.java  -41
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ElasticSearchHealthCheck.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/NotificationDAO.java  -15
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TOSVersion.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java  -35
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/URIHelper.java  -55
- dockstore-webservice/src/main/java/io/openapi/model/ExtendedFileWrapper.java  -41
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/StateManagerMode.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/AbstractDockstoreDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/FileFormat.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedToolResource.java  -30
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/RSSListener.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java  -74
- dockstore-common/src/main/java/io/dockstore/common/Registry.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ServiceDAO.java  -40
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ChecksumConverter.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/TransactionExceptionMapper.java  -91
- dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceConfiguration.java  -84
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/SourceControlResourceInterface.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerFactory.java  -85
- dockstore-common/src/main/java/io/dockstore/common/SourceControl.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Token.java  -38
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  -13
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/LambdaEventResource.java  -47
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceUtilities.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Alias.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/OrganizationDAO.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/UserDAO.java  -26
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/DescriptionSource.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  -7
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceFactory.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java  -61
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ToolDAO.java  -74
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/InMemoryPermissionsImpl.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java  -55
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/LambdaEvent.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AliasHelper.java  -82
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowVersion.java  -75
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResourceDockerRegistries.java  -15
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java  -9
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/BioWorkflow.java  -69
- dockstore-webservice/src/main/java/io/swagger/api/ToolsApiV1.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ServiceEntryDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/BioWorkflowDAO.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ServiceResource.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dockerhub/DockerHubImage.java  -24
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowMode.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java  -80
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java  -75
- dockstore-webservice/src/main/java/io/openapi/api/impl/ToolClassesApiServiceImpl.java  -55
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/PublishRequest.java  -50
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java  -12
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/CollectionDAO.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java  -81
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolClassesApiServiceImpl.java  -5
- dockstore-common/src/main/java/io/dockstore/common/LanguagePluginManager.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/doi/DOIGeneratorFactory.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceControlConverter.java  -72
- dockstore-webservice/src/main/java/io/swagger/model/ExtendedFileWrapper.java  -33
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Event.java  -4
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dockerhub/DockerHubTag.java  -15
- dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguageSubclass.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java  -11
- dockstore-webservice/src/main/java/io/swagger/api/MetadataApiV1.java  -66
- dockstore-webservice/src/main/java/io/swagger/api/impl/MetadataApiServiceImpl.java  -11
- dockstore-common/src/main/java/io/dockstore/common/VersionTypeValidation.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/MetadataResourceHelper.java  -33
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ObsoleteUrlFactory.java  -54
- dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java  -72
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Organization.java  -10
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java  -53
- dockstore-common/src/main/java/io/dockstore/common/LanguageHandlerHelper.java  -80
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java  -82
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/LicenseInformation.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/SitemapListener.java  -75
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/LambdaEventDAO.java  -5
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/StateListenerInterface.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java  -71
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java  -71
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java  -47
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/FileFormatHelper.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TokenTypeConverter.java  -77
- dockstore-common/src/main/java/io/dockstore/common/Utilities.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/Role.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryLabelHelper.java  -91
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsFactory.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Image.java  -67
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/FileDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/VersionMetadata.java  -100
- dockstore-webservice/src/main/java/io/swagger/api/impl/ApiV2BetaVersionConverter.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/PrivacyPolicyVersion.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GoogleHelper.java  -18
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZipSourceFileHelper.java  -4
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApi.java  -22
- dockstore-common/src/main/java/io/dockstore/common/Repository.java  -56
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=6277223&bid=20431333)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903&bid=20290588)

**Pull request merged**